# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1523047052 25200
#      Fri Apr 06 13:37:32 2018 -0700
# Node ID ca20eb3d0535eacd04b290dfde1b87cdd48291aa
# Parent  c7de02b5b2275c9321c2d4319478791f16e0e2ee
ensure pkg tests do not write any files to user's home directory (bug #53557)

* pkg.tst: Change pkg local_list to a temporary file.  Use onCleanup object
to restore pkg local_list when tests are complete.

diff --git a/test/pkg/pkg.tst b/test/pkg/pkg.tst
--- a/test/pkg/pkg.tst
+++ b/test/pkg/pkg.tst
@@ -17,28 +17,31 @@
 ## <https://www.gnu.org/licenses/>.
 
 ############################################################
 ## Test suite for pkg.m
 ## Tests are organized first by action, and then by options.
 ## All actions should be tested, and ideally all options are tested.
 ############################################################
 
-%!shared old_prefix, old_archprefix, prefix, restorecfg, rmtmpdir, mfile_pkg_name, mfile_pkg_zip
+%!shared old_prefix, old_archprefix, old_local_list, prefix, restorecfg, restorecache, rmtmpdir, mfile_pkg_name, mfile_pkg_zip
 %!
 %! ## Do all tests in a temporary directory
 %! [old_prefix, old_archprefix] = pkg ("prefix");
 %! restorecfg = onCleanup (@() pkg ("prefix", old_prefix, old_archprefix));
+%! old_local_list = pkg ("local_list");
+%! restorecache = onCleanup (@() pkg ("local_list", old_local_list));
 %! prefix = tempname ();
 %! [status] = mkdir (prefix);
 %! if (! status)
 %!   error ("pkg.tst: Could not create temporary directory for pkg testing");
 %!   return;  # abort further testing
 %! endif
 %! pkg ("prefix", prefix, prefix);
+%! pkg ("local_list", fullfile (prefix, "octave_packages"));
 %! rmtmpdir = @onCleanup (@() confirm_recursive_rmdir (0, "local") && rmdir (prefix, "s"));
 %!
 %! ## Create zip file packages of testing directories in prefix directory
 %! mfile_pkg_name = {"mfile_basic_test", "mfile_minimal_test"};
 %! mfile_pkg_zip = fullfile (prefix, strcat (mfile_pkg_name, ".zip"));
 %! for i = 1:numel (mfile_pkg_name)
 %!   zip (mfile_pkg_zip{i}, mfile_pkg_name{i});
 %! endfor
